lmp/bb-build.sh

```diff
@@ -8,14 +8,8 @@ start_ssh_agent
 
 source setup-environment build
 
-# Parsing first, to stop in case of parsing issues
-bitbake -p
-
 # Global and image specific envs
 bitbake -e > ${archive}/bitbake_global_env.txt
 bitbake -e ${IMAGE} > ${archive}/bitbake_image_env.txt
 
-# Setscene (cache), failures not critical
-bitbake --setscene-only ${IMAGE} || true
```

Comment:
This was also added to avoid failing the build in case of cache issues.

Commit 5d830195ea4014097211341e17b584071e7357e7.

Reply:
but bitbake will always use the same cache again on the next build until there is a more recent one. I still don't quite understand the problem. if the cache has issues it needs to be invalidated by bitbake rebuild the matching package to generate a new.

Reply:
If we have failing setscene tasks is because our sstate cache is broken and needs to be fixed.
Having --setscene-only doesn't help in this case because if the setscene task fails it will only be updated after running the real task and to do that  we needed to use --no-setscene otherwise bitbake will get the task from the sstate-cache

Reply:
This will revert back to the issue we had before.

Sstate will get fixed by itself at the next run as bitbake will create a valid one as it runs. The problem we have is that bitbake will return with an error when there is a sstate cache issue, even when not really used by bitbake, causing the build to fail (while it was actually successful).

Reply:
what I understand about the sstate cache and the setscene tasks is that the second run will use the same sstate cache as the first because the the task hash is the same in first and in the second.

With the "--setscene-only" bitbake will always fail (when build an image) because afaik the images can't be saved in the sstate-cache so even if he manages to reuse everything will end up failing at the end in the image.

Reply:
> what I understand about the sstate cache and the setscene tasks is that the second run will use the same sstate cache as the first because the the task hash is the same in first and in the second.

At the second run it won't fetch from sstate anymore, and it will assume the one that failed should be created / rebuilt, without causing bitbake to return a value different than 0 (which causes our job to fail).

This was verified as we had an issue before with one of the machines, and after having setscene-only (which was returning an error) run first, the job worked fine.
 
> With the "--setscene-only" bitbake will always fail (when build an image) because afaik the images can't be saved in the sstate-cache so even if he manages to reuse everything will end up failing at the end in the image.

For this I would have to confirm, but not an issue with the current logic because we do have || true (but I believe it will just try to restore whatever is possible and not really try to build anything).

Reply:
The current logic works as is and my patch is minor improvement that run only one less task so will close the PR because it doesn't really add much